src/Symfony/Component/Routing/Generator/UrlGenerator.php

```diff
@@ -157,9 +157,8 @@ protected function doGenerate($variables, $defaults, $requirements, $tokens, $pa
         foreach ($tokens as $token) {
             if ('variable' === $token[0]) {
                 $varName = $token[3];
-                if ($important = ('!' === $varName[0])) {
-                    $varName = substr($varName, 1);
-                }
+                // variable is not important by default
+                $important = $token[5] ?? false;
```

Comment:
I guess it's out of the scope of this PR, but I really dislike the "important" name. Having an important/non-important route variable doesn't tell me exactly what's its purpose. It may be better to use mandatory/optional, included/optional, obligatory/optional, required/optional, etc.

Reply:
I think, this is the parallel with css `!important` rule. While it does not tell the purpose, it gives a strong association with a prepended `!` :-)
For url generation parameter is still optional, while for generated url it is ..?
Probably, `$generatedUrlIncluded`?

Reply:
With #29763, it's not going to be used only for generation anymore.
You're correct about "important" being inspired by CSS.
I proposed the name so I'm fine with it really :)